sql: move dist_sql to separate package

RaduBerinde opened the pull request:
Moving dist_sql APIs to separate package. Adding `TestMain` to set up the
testing shim.

Renaming files and `DistSQL*` types to avoid stuttering.

Fixing `protobuf.mk` to only generate the extra stuff for the protos in roachpb.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6561)

<!-- Reviewable:end -->

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/sqlbase/expr_filter.go, line 23 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6561#-KH6Icl3IEBUPLQBz63h:-KH6Icl4sNSak0mgj-Eb:-346202470) ([raw file](https://github.com/cockroachdb/cockroach/blob/a637e87750bc599f570e6f3a791d6744d89b1573/sql/sqlbase/expr_filter.go#L23)):*
I wonder if this function really needs to exist in base, with its own file... It's really just calling `filter.Eval()`. We could duplicate this code.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6561#-:-KH6JIQN1NlleSfUiz7B:1417088721)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

---

Review status: 0 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/sqlbase/expr_filter.go, line 23 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6561#-KH6Icl3IEBUPLQBz63h:-KH6K35puPvgbGfBq57-:538073217) ([raw file](https://github.com/cockroachdb/cockroach/blob/a637e87750bc599f570e6f3a791d6744d89b1573/sql/sqlbase/expr_filter.go#L23)):*
Sure but what's the downside of it being in sqlbase? Better than duplicating it IMO

As far as the file is concerned, we will probably need other expression-related functions usable from both sql and distsql. If not, we can probably move this to parser (as a more generic `EvalBoolExpr` or something).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6561#-:-KH6KXXNwpGp_NdAETsY:-485616692)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: move dist_sql to separate package

Moving dist_sql APIs to separate package. Adding `TestMain` to set
up the testing shim.

Renaming files and `DistSQL*` types to avoid stuttering.

Fixing `protobuf.mk` to only generate the extra stuff for the protos
in `roachpb`.

RaduBerinde committed:
sql: switch table reader test to use the new testingshim

Adding utility functions that parallel the existing ones in sql_test, but which
use the testingshim. Switching the table reader test to use it; this is a
prequel to moving the dist_sql stuff to a separate package.

Also moved getTableLease to sqlbase; this will need to happen for any
functionality that is used for both sql and dis_sql tests.

RaduBerinde committed:
sql: move RunFilter to sqlbase
